design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+# Stacks Security and Isolation
+
+* Owner: Jared Watts (@jbw976)
+* Reviewers: Crossplane Maintainers
+* Status: Draft, revision 1.0
+
+## Background
+
+The current model for Stacks, the means of extending Crossplane with new functionality, allows the stack author to declare arbitrary permissions that the stack will be granted by the Stack Manager.
+If the Stack author declares they need to update nodes or create volumes, those permissions will be granted.
+
+However, Crossplane is not an arbitrary Kubernetes cluster to run typical compute and storage workloads.
```

Comment:
I am thinking it may be worth calling out that Crossplane can still run within/alongside an arbitrarily used Kubernetes cluster, even though Crossplane itself is not intended to support arbitrary extensibility. Without saying it, readers may think that running Crossplane alongside other Kubernetesing may not be a supported use-case.

Reply:
Is a Stack that installs and manages (for example) Wordpress on the cluster upon which Crossplane is running (i.e. via a Deployment, not a KubernetesApplication) a use case we want to support? We've wavered on this point over the last few months. 

My current understanding is that deploying Crossplane in order to use the services end of our functionality is a pattern we bless and encourage. i.e. As an operator of an existing cluster, I can install Crossplane and stack-gcp in order to allow my cluster's users to manage their GCP infrastructure via classes, claims, and managed resources. If I then wanted to provide my users access to (for example) Jenkins, would I be required to create a `KubernetesCluster` and `KubernetesApplication` to run Jenkins?